File: app/src/main/java/com/matchthree/game/model/Position.kt
package com.matchthree.game.model

import kotlin.math.abs

/** A cell coordinate on the board. */
data class Position(val row: Int, val col: Int) {
    /** True if [other] is one step up, down, left, or right of this position. */
    fun isOrthogonallyAdjacentTo(other: Position): Boolean =
        (row == other.row && abs(col - other.col) == 1) ||
            (col == other.col && abs(row - other.row) == 1)
}

File: Engine/Model/Position.cs
namespace MatchThree.Engine.Model;

public readonly record struct Position(int Row, int Col)
{
    public bool IsOrthogonallyAdjacentTo(Position other) =>
        (Row == other.Row && Math.Abs(Col - other.Col) == 1) ||
        (Col == other.Col && Math.Abs(Row - other.Row) == 1);
}